### PR TITLE
HAI-1526 Add functions to send hanke and application invitation emails

### PR DESCRIPTION
**Pull request by pitkni**
# Description

Add templates for hanke and application invitation emails. Add functionality to send these invitations in EmailSenderService.

Note: It is possible for inviter name to be unknown. In these cases the emails use only the inviter email (foo.bar@mail.fi kutsui sinut..).

Note: mjml.io plugin is used to create the email templates. From mjml documentation:

MJML is a markup language designed to reduce the pain of coding a responsive email. Its semantic syntax makes it easy and straightforward and its rich standard components library speeds up your development time and lightens your email codebase. MJML’s open-source engine generates high quality responsive HTML compliant with best practices.

Note: Email translations are not included in this pr, as mentioned in the Jira ticket.

Using the new email sending features will be implemented on a different upcoming pr.

## Type of change

- [ ] Bug fix 
- [x] New feature 
- [ ] Other

# Instructions for testing
This cannot be directly tested, but adding something like this to EmailSenderService can be used for verification:

```

    @Scheduled(fixedDelay = 60000)
    fun testSendHanke() {
        sendHankeInvitationEmail(
            to = "something@mail.com",
            HankeInvitationData(
                inviterName = "Matti Meikäläinen",
                inviterEmail = "foo@bar.com",
                hankeTunnus = "HAI24-1",
                hankeNimi = "Testi hanke",
                invitationToken = "testtoken"
            )
        )
    }

    @Scheduled(fixedDelay = 60000)
    fun testSendHakemus() {
        sendApplicationInvitationEmail(
            to = "something@mail.com",
            ApplicationInvitationData(
                inviterName = "Matti Meikäläinen",
                inviterEmail = "foo@bar.com",
                applicationType = ApplicationType.CABLE_REPORT,
                applicationIdentifier = "JS002",
                hankeTunnus = "HAI24-1",
                roleType = ContactType.RAKENNUTTAJA,
            )
        )
    }
```
Then you can check localhost:3003 (email server in docker compose)

# Checklist:

- [x] I have written new tests (if applicable)
- [x] I have ran the tests myself (if applicable)
- [ ] I have made necessary changes to the documentation, link to confluence
 or other location: 

# Other relevant info
Please describe here if there is e.g. some requirements for this change or
 other info that the tester/user needs to know.

### CODE REVIEW

File: services/hanke-service/src/main/resources/email/template/attributes.mjml
Comment (corvidian) [inline services/hanke-service/src/main/resources/email/template/attributes.mjml:1]: The mjml files could be moved elsewhere. Under resources, they will be a part of the runtime package, which they clearly don't need to be.

File: services/hanke-service/src/main/resources/email/template/johtoselvitys-valmis.html.mustache
Comment (corvidian) [inline services/hanke-service/src/main/resources/email/template/johtoselvitys-valmis.html.mustache:1]: Maybe we should keep this file as the old HTML version until we get approval for the new design. We can keep the mjml-file, just not compile it until we're ready to deploy it to production.

File: README.md
Comment (corvidian) [inline README.md:167]: The IntelliJ IDEA plugin can't do the HTML export. We'll need more tooling to make the templates ergonomic, but that can be tackled later.

File: services/hanke-service/src/main/resources/email/template/attributes.mjml
Comment (pitkni) [reply services/hanke-service/src/main/resources/email/template/attributes.mjml:1]: Do you have any suggestions on where they should be located?

Comment (pitkni) [reply services/hanke-service/src/main/resources/email/template/attributes.mjml:1]: How about an emailtemplate directory under scripts?

File: README.md
Comment (pitkni) [reply README.md:167]: IntelliJ plugin provides the html output that you can copy-paste. I did not want to write very detailed instructions here as we should probably have a script etc. to automate this (which is tackled later).

File: services/hanke-service/src/main/resources/email/template/johtoselvitys-valmis.html.mustache
Comment (pitkni) [reply services/hanke-service/src/main/resources/email/template/johtoselvitys-valmis.html.mustache]: Reverted.

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/email/EmailSenderService.kt
Comment (pitkni) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/email/EmailSenderService.kt]: Yes, technically it chooses the template. I wanted to use the word context to improve the readability, for example that "now we are sending a hanke invitation".  Changed the class name to EmailTemplate.

File: services/hanke-service/src/main/resources/email/template/attributes.mjml
Comment (corvidian) [reply services/hanke-service/src/main/resources/email/template/attributes.mjml:1]: They're not scripts, so that doesn't feel correct. Emails directory in either project root or `services/hanke-service/emails`?

Comment (pitkni) [reply services/hanke-service/src/main/resources/email/template/attributes.mjml]: Placed them in project root email/.

Comment (corvidian) [reply services/hanke-service/src/main/resources/email/template/attributes.mjml]: 👍 Dividing the common parts to a separate directory was a good addition.

**Review by corvidian — APPROVED** [review]

**Pull request merged**
